app/src/main/java/com/example/abhishek/bookshareapp/ui/adapter/BooksAdapter.java

```diff
@@ -1,119 +1,91 @@
 package com.example.abhishek.bookshareapp.ui.adapter;
 
 import android.content.Context;
+import android.support.v7.widget.RecyclerView;
 import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.BaseAdapter;
 import android.widget.ImageView;
 import android.widget.TextView;
 import android.widget.Toast;
 
 import com.example.abhishek.bookshareapp.R;
 import com.example.abhishek.bookshareapp.api.models.Book;
-import com.example.abhishek.bookshareapp.ui.SearchResultsActivity;
 import com.squareup.picasso.Picasso;
 
 import java.util.List;
 
-/**
- * Created by abhishek on 30/1/16.
- */
-public class BooksAdapter extends BaseAdapter implements View.OnClickListener {
+public class BooksAdapter extends RecyclerView.Adapter<BooksAdapter.ViewHolder>{
 
     private Context context;
     private List<Book> bookList;
     private static LayoutInflater inflater=null;
     Book tempValues=null;
 
-    public BooksAdapter(Context context,List<Book> bookList){
-        this.bookList =bookList;
-        this.context=context;
-        inflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
-        Log.d("BookAdapter","Constructor");
-    }
-
-
-
-    @Override
-    public Object getItem(int position) {
-        return bookList.get(position);
-    }
-
-    @Override
-    public long getItemId(int position) {
-        return 0;
-    }
-
-    public static class ViewHolder{
+    public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
         public TextView title_book;
         public TextView author_book;
         public ImageView image_book;
+        Context context;
+
+        public ViewHolder(View v, Context context){
+            super(v);
+            title_book = (TextView)v.findViewById(R.id.row_books_title);
+            author_book = (TextView)v.findViewById(R.id.row_books_author);
+            image_book = (ImageView) v.findViewById(R.id.row_books_imageView);
+            title_book.setOnClickListener(this);
+            author_book.setOnClickListener(this);
+            image_book.setOnClickListener(this);
+            this.context = context;
+        }
+
+        @Override
+        public void onClick(View v) {
+            Toast.makeText(context,"Clicked",Toast.LENGTH_SHORT).show();
+        }
     }
 
-    @Override
-    public int getCount() {
-        return bookList.size();
+    public BooksAdapter(Context context,List<Book> bookList){
+        this.bookList =bookList;
```

Comment:
...while `CamelCase` here. Is there a specific reason ? If no, please make it uniform throughout :)